### DISCUSSION TITLE
K3s + METALLB + UDP = checksum errors?

### DISCUSSION THREAD

**Discussion by m4w0lf** [k3s-io/k3s#7675] [Q&A]
Hi everyone, 
I'm currently facing a very particular issue involving udp traffic and services.

**Environment:**
Debian (bullseye) kernel -> 5.19.11-1~bpo11+1
K3s v1.26.5+k3s1

host physical ip: 10.162.5.10
svc external ip: 10.162.255.9

SVC config:
```console
root@Hydra:~# kubectl  describe svc -n infra-pxeboot      tftpd        
Name:                     tftpd
Namespace:                infra-pxeboot
Labels:                   app=pxeboot
                          app.kubernetes.io/component=tftpd
                          app.kubernetes.io/instance=pxeboot
                          app.kubernetes.io/managed-by=Helm
                          app.kubernetes.io/name=pxeboot
                          app.kubernetes.io/version=1.0.0
Annotations:              meta.helm.sh/release-name: pxeboot
                          meta.helm.sh/release-namespace: infra-pxeboot
                          metallb.universe.tf/allow-shared-ip: 10.162.255.9
                          metallb.universe.tf/ip-allocated-from-pool: dc-mgmt-infra-pool1
                          metallb.universe.tf/loadBalancerIPs: 10.162.255.9
Selector:                 app=pxeboot
Type:                     LoadBalancer
IP Family Policy:         SingleStack
IP Families:              IPv4
IP:                       6.1.191.30
IPs:                      6.1.191.30
LoadBalancer Ingress:     10.162.255.9
Port:                     tftpd  69/UDP
TargetPort:               tftpd/UDP
NodePort:                 tftpd  30057/UDP
Endpoints:                6.0.0.162:69
Session Affinity:         None
External Traffic Policy:  Local
HealthCheck NodePort:     30042
```

**Issue:**
traffic directed to the service is correcly handled and delivered to the pod.
response traffic from the pod seems to be dropped somewhere between pod veth and physical nic of the host.
digging around with tcpdump it's kinda confirming this theory: response packets from POD have the famous "incorrect checksum" warning.

the crazy fact is that the wrong checksum is happening only when using a "virtual" external ip. binding the service to the physical host solves the checksum error.

i already tried all "disable checksum offload" ethtool dark magic, (on physical and cni0 nic) but nothing changed.

do you have any advice on further investigations?

Thank you and best regards,
Alex

**Comment by kim0** [comment]
@m4w0lf Hello .. I think I'm being hit by this same bug. Did you ever discover something useful about it? Thanks!
